src/rust/catnip/interop.rs

```diff
@@ -39,12 +39,20 @@ pub fn pack_result(rt: Rc<DPDKRuntime>, result: OperationResult, qd: QDesc, qt:
             qr_qt: qt,
             qr_value: unsafe { mem::zeroed() },
         },
-        OperationResult::Accept(new_qd) => {
-            let sin = unsafe { mem::zeroed() };
-            let qr_value = demi_qr_value_t {
+        OperationResult::Accept((new_qd, addr)) => {
+            let saddr: SockAddrIn = {
+                // TODO: check the following byte order conversion.
```

Comment:
I thought I'd do this TODO as part of this review.

I believe the byte-order here is inadvertently correct.  `addr` is a `std::net::SocketAddrV4`, whose `ip()` method returns a `std::net::Ipv4Addr`, whose `octets()` method returns a `[u8; 4]` where the order of the bytes contained therein is **big-endian**.  Line 48's `pal::create_sin_addr()` function appears to expect its input bytes in **little-endian** byte order.

This still works (but only on our little-endian machines!) because `create_sin_addr()` is buggy.  As written, it creates a native byte order u32 from little-endian bytes.  What it **should** be doing is creating a network (i.e., big-endian) byte order u32 from big-endian bytes.  In other words, it shouldn't reverse the byte order.  And on little-endian machines, where the native byte order is (no surprise) little-endian, creating a native byte order u32 from little-endian bytes doesn't reverse the byte order either.  So `create_sin_addr()`'s bug doesn't manifest itself on little-endian machines.  But on a big-endian machine, it would reverse the byte order, which would be wrong.

I'll file an Issue for the `create_sin_addr()` bug.  This code is correct.  So, you can remove this TODO comment (and the identical one on line 75 you presumably copied this from).

Reply:
Sure.